lib/puppet/parser/compiler.rb

```diff
@@ -31,7 +31,7 @@ def self.compile(node)
       raise(Puppet::Error, errmsg.join(' '))
     end
 
-    new(node).compile.to_resource
+    new(node).compile {|resulting_catalog| resulting_catalog.to_resource }
```

Comment:
@hlindberg could you add a test to confirm that `to_resource` was performed in the context of the *correct* environment? Perhaps check that the compiler generated resources have the correct environment set on them?